rsocket-core/src/commonTest/kotlin/io/rsocket/kotlin/frame/io/CustomMimeTypeTest.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket.kotlin.frame.io
+
+import io.ktor.utils.io.core.*
+import io.rsocket.kotlin.core.*
+import kotlin.test.*
+
+class CustomMimeTypeTest {
```

Comment:
let's rename to just `MimeTypeTest` as I was planning to add such test soon :)

Reply:
Also, we do have `TestWithLeakCheck` and helper builders in [Packets.kt](https://github.com/rsocket/rsocket-kotlin/blob/2239c5ca176694effbd803643d0179e79f76be3f/rsocket-test/src/commonMain/kotlin/io/rsocket/kotlin/test/Packets.kt) (like `packet {...}` which will use custom leak-tracing pool to validate, that there is no leaks in tests - so better to use it here

Reply:
yup, changed as suggested